hw/top_earlgrey/syn/asic.constraints.sdc

```diff
@@ -4,205 +4,231 @@
 
-## TODO: Create generated clock for negedge SPID_CLK. Then make them clock group
+## TODO: these are dummy constraints and likely incorrect, need to properly constrain min/max
+set SPI_HOST_IN_DEL_FRACTION 0.7
+set SPI_HOST_OUT_DEL_FRACTION 0.7
+set SPI_HOST_IN_DEL    [expr ${SPI_HOST_IN_DEL_FRACTION} * ${SPI_HOST_TCK}]
+set SPI_HOST_OUT_DEL   [expr ${SPI_HOST_OUT_DEL_FRACTION} * ${SPI_HOST_TCK}]
 
-set IN_DEL    6.0
-set OUT_DEL   6.0
+# bidir ports
+set_input_delay ${SPI_HOST_IN_DEL} [get_ports SPI_HOST_CS_L] -clock SPI_HOST_CLK
+set_input_delay ${SPI_HOST_IN_DEL} [get_ports SPI_HOST_D0]   -clock SPI_HOST_CLK
+set_input_delay ${SPI_HOST_IN_DEL} [get_ports SPI_HOST_D1]   -clock SPI_HOST_CLK
+set_input_delay ${SPI_HOST_IN_DEL} [get_ports SPI_HOST_D2]   -clock SPI_HOST_CLK
+set_input_delay ${SPI_HOST_IN_DEL} [get_ports SPI_HOST_D3]   -clock SPI_HOST_CLK
 
-set_input_delay ${IN_DEL} [get_ports dio_in_core[$PORT_SPI_DEVICE_CSB]]   -clock SPID_CLK
-set_input_delay ${IN_DEL} [get_ports dio_in_core[$PORT_SPI_DEVICE_S0]]    -clock SPID_CLK
-set_input_delay ${IN_DEL} [get_ports dio_in_core[$PORT_SPI_DEVICE_S1]]    -clock SPID_CLK
-set_input_delay ${IN_DEL} [get_ports dio_in_core[$PORT_SPI_DEVICE_S2]]    -clock SPID_CLK
-set_input_delay ${IN_DEL} [get_ports dio_in_core[$PORT_SPI_DEVICE_S3]]    -clock SPID_CLK
+set_output_delay ${SPI_HOST_OUT_DEL} [get_ports SPI_HOST_CS_L] -clock SPI_HOST_CLK
+set_output_delay ${SPI_HOST_OUT_DEL} [get_ports SPI_HOST_D0]   -clock SPI_HOST_CLK
+set_output_delay ${SPI_HOST_OUT_DEL} [get_ports SPI_HOST_D1]   -clock SPI_HOST_CLK
+set_output_delay ${SPI_HOST_OUT_DEL} [get_ports SPI_HOST_D2]   -clock SPI_HOST_CLK
+set_output_delay ${SPI_HOST_OUT_DEL} [get_ports SPI_HOST_D3]   -clock SPI_HOST_CLK
 
-set_output_delay ${OUT_DEL} [get_ports dio_out_core[$PORT_SPI_DEVICE_S0]] -clock SPID_CLK
-set_output_delay ${OUT_DEL} [get_ports dio_oe_core[$PORT_SPI_DEVICE_S0]]  -clock SPID_CLK
-set_output_delay ${OUT_DEL} [get_ports dio_out_core[$PORT_SPI_DEVICE_S1]] -clock SPID_CLK
-set_output_delay ${OUT_DEL} [get_ports dio_oe_core[$PORT_SPI_DEVICE_S1]]  -clock SPID_CLK
-set_output_delay ${OUT_DEL} [get_ports dio_out_core[$PORT_SPI_DEVICE_S2]] -clock SPID_CLK
-set_output_delay ${OUT_DEL} [get_ports dio_oe_core[$PORT_SPI_DEVICE_S2]]  -clock SPID_CLK
-set_output_delay ${OUT_DEL} [get_ports dio_out_core[$PORT_SPI_DEVICE_S3]] -clock SPID_CLK
-set_output_delay ${OUT_DEL} [get_ports dio_oe_core[$PORT_SPI_DEVICE_S3]]  -clock SPID_CLK
+#####################
+# SPI passthrough   #
+#####################
+
+# input pad + internal + output pad
+set TPAD_I 1.2
+set THODI  2.0
+set TPAD_O 3.3
+set SPI_HODI_PASS_MAX_DELAY [expr ${TPAD_I} + ${THODI} + ${TPAD_O}]
+set SPI_HIDO_PASS_MAX_DELAY ${SPI_HODI_PASS_MAX_DELAY}
+
+# TODO: These are strawman constraints and need to be refined.
+set_max_delay ${SPI_HODI_PASS_MAX_DELAY} -from [get_ports SPI_DEV_D0] -to [get_ports SPI_HOST_D0]
```

Comment:
it might be better to have a `-thru` here for the muxes @eunchan specifically placed for passthrough.

Reply:
we can address that later though.